app/src/main/java/com/example/advanceduiapp/PagerActivity.kt

```diff
@@ -1,12 +1,63 @@
 package com.example.advanceduiapp
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import com.example.advanceduiapp.R
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.viewpager.widget.ViewPager
+import com.example.advanceduiapp.data.PageData
+import com.example.advanceduiapp.databinding.ActivityPagerBinding
+import com.example.advanceduiapp.ui.pager.PagerFragment
+import com.example.advanceduiapp.ui.pager.PagerViewModel
 
 class PagerActivity : AppCompatActivity() {
+
+    private lateinit var pager: ViewPager
+    private lateinit var binding: ActivityPagerBinding
+    private val viewModel: PagerViewModel by viewModels()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_pager)
+        binding = ActivityPagerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        supportActionBar?.hide()
+
+        pager = findViewById(R.id.view_pager)
+        val pagerAdapter = ImagePagerAdapter(supportFragmentManager, viewModel.imageMap)
```

Comment:
you have to observe the list of items in viewModel and than update the viewPager. Let's just image that you get the `imageMap` from the backend